RIOT-OS/RIOT
# sys/registry: Add File storage

### Contribution description
This PR adds an File based storage facility for the RIOT Registry (#10799). This allows to save the configuration parameters in an File of an available file system.

It also modifies the tests/registry application to easily integrate this storage. In the test the VFS and FAT modules are used.

### Testing procedure
Modify the Makefile of the tests/registry application to use the registry_store_file instead of the registry_store_dummy. Run the application, modify the values and save them. They should persist after rebooting the node.

### Issues/PRs references
Depends on PR #10799.
